### PR TITLE
fix: install deps for e2e tests in ci

### PR DESCRIPTION
**Pull request by MCarlomagno**
# Summary
Adds step in CI for installing e2e folder dependencies

### CODE REVIEW

File: .github/actions/prepare/action.yml [hunk 1]
```diff
@@ -24,3 +24,8 @@ runs:
       run: |
         pnpm install-deps
       shell: bash
+
+    - name: Install e2e dependencies
+      run: |
+        cd e2e && pnpm install
```
Comment (tirumerla) [inline .github/actions/prepare/action.yml:30]: You probably need to use frozen lock file instead of updating the lock file, Similar to install-deps.

**Review by tirumerla — CHANGES_REQUESTED** [review]
added a comment, thanks

File: package.json [hunk 2]
```diff
@@ -36,16 +36,16 @@
     "ci:version": "pnpm changeset version",
     "ci:snapshot": "pnpm changeset version --snapshot rc-snapshot",
     "sort:networks": "node hack/sortNetworks.js && prettier -u -w packages/base/src/utils/network*.ts",
-    "build": "nx run-many -t build --parallel=1",
-    "build-test": "nx run-many -t style,build,test --projects=@openzeppelin/defender-base-client,*  --parallel=false",
+    "build": "nx run-many -t build --parallel=1 --output-style",
```
Comment (tirumerla) [inline package.json:39]:
```suggestion
    "build": "nx run-many -t build --parallel=1 --output-style=static",
```

File: package.json [hunk 3]
```diff
@@ -36,16 +36,16 @@
...
+    "build-test": "nx run-many -t style,build,test --projects=@openzeppelin/defender-base-client,*  --parallel=false --output-style",
```
Comment (tirumerla) [inline package.json:40]:
```suggestion
    "build-test": "nx run-many -t style,build,test --projects=@openzeppelin/defender-base-client,*  --parallel=false --output-style=static",
```

File: package.json [hunk 4]
```diff
@@ -36,16 +36,16 @@
...
     "lint:check": "eslint 'packages/**/src/**/*.{js,ts}' --quiet",
     "lint:fix": "pnpm prettier:fix && pnpm lint:check && pnpm prettier:check",
     "prettier:check": "prettier -u --check '**/*.{js,ts,tsx,md}' '!**/.nx/**'",
     "prettier:fix": "prettier -u --write '**/*.{js,ts,tsx,md}' '!**/.nx/**'",
-    "test": "nx run-many -t test --exclude=e2e --parallel=1",
-    "test:e2e": "nx run e2e:test",
-    "build:changed": "nx affected:build --base=origin/main --skip-nx-cache --parallel=1",
-    "test:changed": "nx affected:test --base=origin/main --skip-nx-cache --parallel=1",
+    "test": "nx run-many -t test --exclude=e2e --parallel=1 --output-style",
```
Comment (tirumerla) [inline package.json:45]:
```suggestion
    "test": "nx run-many -t test --exclude=e2e --parallel=1 --output-style=static",
```

File: package.json [hunk 5]
```diff
@@ -36,16 +36,16 @@
...
+    "test:e2e": "nx run e2e:test --output-style=static",
+    "build:changed": "nx affected:build --base=origin/main --skip-nx-cache --parallel=1 --output-style",
```
Comment (tirumerla) [inline package.json:47]:
```suggestion
    "build:changed": "nx affected:build --base=origin/main --skip-nx-cache --parallel=1 --output-style=static",
```

File: package.json [hunk 6]
```diff
@@ -36,16 +36,16 @@
...
+    "test:changed": "nx affected:test --base=origin/main --skip-nx-cache --parallel=1 --output-style",
```
Comment (tirumerla) [inline package.json:48]:
```suggestion
    "test:changed": "nx affected:test --base=origin/main --skip-nx-cache --parallel=1 --output-style=static",
```

**Review by tirumerla — APPROVED** [review]
lgtm, thanks

**Review by shahnami — APPROVED** [review]

**Pull request merged**